system/ee/ExpressionEngine/Controller/Homepage/Homepage.php

```diff
@@ -212,6 +217,20 @@ public function dismissBanner()
 
         ee()->output->send_ajax_response(['success']);
     }
+
+    /**
+     * Dismiss post-update banner from CP homepage, until next update
+     *
+     * @return void
+     */
+    public function switchTheme()
+    {
+        $member = ee()->session->getMember();
+        $member->cp_theme = ee('Request')->isPost() ? ee('Request')->post('theme') : ee('Request')->get('theme');
```

Comment:
Can we validate theme is in an array of allowed themes?

Reply:
The validation is done in javascript file